update/update.go

```diff
@@ -24,6 +24,15 @@ type StateEntry struct {
 
 // CheckForUpdate checks whether this software has had a newer release on GitHub
 func CheckForUpdate(client *api.Client, stateFilePath, repo, currentVersion string) (*ReleaseInfo, error) {
+	stateEntry, err := getStateEntry(stateFilePath)
```

Comment:
It seems that the caller doesn't check the returned error: 
https://github.com/cli/cli/blob/34d549e7b61660c7c993181c0be046d6277cad03/cmd/gh/main.go#L231-L235
https://github.com/cli/cli/blob/34d549e7b61660c7c993181c0be046d6277cad03/cmd/gh/main.go#L38-L41

so this will either crash at runtime or left a new user in a loop where it never checks for updates since `state.yml` will never be generated by `getLatestReleaseInfo` https://github.com/cli/cli/blob/34d549e7b61660c7c993181c0be046d6277cad03/update/update.go#L51


Dont merge this, I'll test it again and will update if necessary.